### PR TITLE
Init changelog 24.02 [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Init changelog with

    GITHUB_TOKEN=<> scripts/generate-changelog --releases=23.12,24.02

    move 23.10 content to archives folder, we only retain 2 months' GitHub changelog

### CODE REVIEW

**Comment by NvTimLiu** [comment]
Please click the Load diff to see the new changelog changes.

this is for in-advance changelog review, we will re-generate multiple times to update content to latest later,
@mattahrens @sameerz @GaryShen2008 to help adjust issues and changes thanks

**Comment by sameerz** [comment]
Reviewed the changelog and made some minor changes to issues.  I am good after it is regenerated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -572,8 +578,6 @@ Generated on 2024-02-06
 |[#9373](https://github.com/NVIDIA/spark-rapids/pull/9373)|Fix auto merge conflict 9372|
 |[#9308](https://github.com/NVIDIA/spark-rapids/pull/9308)|Initiate arm64 CI support [skip ci]|
 |[#9292](https://github.com/NVIDIA/spark-rapids/pull/9292)|Init project version 23.12.0-SNAPSHOT|
-|[#9291](https://github.com/NVIDIA/spark-rapids/pull/9291)|Automerge from 23.10 to 23.12 [skip
```
Comment (NvTimLiu) [inline CHANGELOG.md]: Updated, but I'm curious why this line is removed from the CHANGELOG history

**Review by jlowe — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (NvTimLiu) [reply CHANGELOG.md]: Oh it's target to branch-23.10

**Review by sameerz — DISMISSED** [review]

**Comment by NvTimLiu** [comment]
build

**Review by GaryShen2008 — APPROVED** [review]

**Pull request merged**